### DISCUSSION TITLE
How are recent languages stats calculated

### DISCUSSION THREAD

**Discussion by Nixinova** [lowlighter/metrics#332] [Q&A] [answered]
Does the recent languages plugin measure:
1. how many recent commits feature that language,
2. the total number of bytes changed for each file type in recent commits, or
3. the total file size of all files recently touched?

**Comment by lowlighter** [comment, marked as answer]
1. number of commits/files/days used to compute these stats are now displayed since 4747332. It can be customized through `plugin_languages_recent_load` (only *PushEvents* will be kept from loaded events) and `plugin_languages_recent_days` options
2. total number of bytes changed is available when using *bytes-size* from `plugin_languages_details` option. these are obtained by filtering added lines for each diff patch and applying `Buffer.byteLength` on them (assuming encoding is utf-8)
3. total file size is available computed though it's not displayed (it's currently only used to compute percentages)

<img src="https://user-images.githubusercontent.com/22963968/119811663-a29f3e00-bee7-11eb-883f-645c5e4f5d34.png">

Seems that some languages are mistaken for others though. As you can see it deduced that I recently did a lot of HTML, but it was actually EJS. It's coherent with what I get when running it locally:
```
GitHub/metrics> github-linguist
62.22%  JavaScript
33.54%  HTML
4.03%   CSS
0.21%   Dockerfile
```

Edit: actually it seems to be misdetected only on WSL, the GitHub runner seems to correctly guess languages

**Reply by Nixinova** [reply]
So if I have say 95% JavaScript that means 95% of the bytes changed in the past 2 weeks by me were JS?

**Reply by lowlighter** [reply]
Yeah in theory that's what it should represent

I don't know how accurate it is since it's been introduced recently, but it seems ok 
If you find any numbers that seems off, don't hesistate to report it 👍
